src/syscalls/main.rs

```diff
@@ -0,0 +1,293 @@
+use bcc::core::BPF;
+use bcc::BccError;
+use clap::{App, Arg, ArgMatches};
+
+use core::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::{mem, ptr, thread, time};
+
+mod syscall;
+
+// A tool for reporting syscall count and latency
+//
+// Based on: https://github.com/iovisor/bcc/blob/master/tools/syscount.py
+
+#[repr(C)]
+struct data_key_t {
+    count: u64,
+    total_ns: u64,
+}
+
+const MILLISECOND: u64 = 1_000 * MICROSECOND;
+const MICROSECOND: u64 = 1_000 * NANOSECOND;
+const NANOSECOND: u64 = 1;
+
+fn get_code(matches: &ArgMatches) -> String {
+    let mut code = include_str!("bpf.c").to_string();
+    if matches.is_present("pid") {
+            let pid: u32 = matches
+            .value_of("pid")
+            .unwrap()
+            .parse()
+            .expect("Invalid pid");
+        
+        code = format!("#define FILTER_PID {} \n {}", pid, code);
+    }
+    if matches.is_present("failures") {
+        code = format!("#define FILTER_FAILED {}", code);
+    }
+    if matches.is_present("errorno") {
+        let errorno = matches.value_of("errorno").unwrap();
+        code = format!("#define FILTER_ERRNO {} \n {}", errorno, code)
+    }
+    if matches.is_present("latency") {
+        code = format!("#define LATENCY \n {}", code);
+    }
+
+    code
+}
+
+fn do_main(runnable: Arc<AtomicBool>) -> Result<(), BccError> {
+    let matches = App::new("syscalls")
+        .about("Summarize number of syscalls and their latencies")
+        .arg(
+            Arg::with_name("interval")
+                .long("interval")
+                .value_name("Seconds")
+                .help("Integration window duration and period for stats output")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("windows")
+                .long("windows")
+                .value_name("Count")
+                .help("The number of intervals before exit")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("milliseconds")
+                .long("milliseconds")
+                .short("M")
```

Comment:
Do these match the python? Seems odd to have `M` for milliseconds and `m` for microseconds when the SI prefix would be `m` and `u` correspondingly.